lisa/tools/reboot.py

```diff
@@ -21,22 +19,19 @@ def _check_exists(self) -> bool:
         return True
 
     def reboot(self) -> None:
-        uptime = self.node.tools[Uptime]
+        who = self.node.tools[Who]
         timer = create_timer()
-        before_reboot_since_time = uptime.since_time()
-        current_since_time = before_reboot_since_time
-        self._log.debug(f"rebooting with current uptime: {before_reboot_since_time}")
-        self.node.execute(f"sudo {self.command}")
-        self.node.shell.close()
+        last_boot_time = who.last_boot()
```

Comment:
Hey, just a heads up, this doesn't work on WSL (`who` never returns anything on WSL, it's probably depending on systemd or something, which WSL doesn't have, but `uptime` I think just reads from `/proc` so it works).

Reply:
uptime doesn't work on centos 6.x, since it doesn't support `--since` parameter, which returns an absolute start time. /proc/uptime also is a relative seconds, not reliable to detect a restart. Another choice is `last`, both `last` and `who` are in centos 6.x. My Ubuntu 2004 WSL doesn't support reboot. If there is other requirements, it can be switched to `last`.